### ISSUE TITLE
semantic!: Remove `SymbolTable::get_symbol_id_from_name` and `SymbolTable::get_scope_id_from_name`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#5456]
These methods are in my opinion a footgun because they don't do what you might think they do. Worse still, minimal test cases often won't reveal that.

We use `get_symbol_id_from_name` to get the `SymbolId` for a reference, but actually that's not what it's doing - it will give you the `SymbolId` of the *first* binding with this variable name *anywhere* in the AST.

They are not the same. e.g. in this case:

```js
{
    let foo;
}
let bar = foo;
```

If we want to know if `foo` in `let bar = foo;` refers to a local variable, `get_symbol_id_from_name("foo").is_some()` says "yes it does". But that's wrong.

Instead we should use `symbol_table.is_global_reference(ident.reference_id().unwrap())`.

As well as being correct, this is also much more performant, as it doesn't iterate through every single symbol in the entire AST the way `get_symbol_id_from_name` does.

#### `get_symbol_id_from_name`

* #5446 removes a usage and adds tests which demonstrate it was incorrect.
* #5450 removed another one.
* The only remaining usage of is in the `tree_shaking/no_side_effects_in_initialization` linter rule, and I believe it's incorrect there too (#5455).

#### `get_scope_id_from_name`

This method is not used anywhere in Oxc.

**Comment by overlookmotel** [comment]
#5467 removed the last remaining use of `get_symbol_id_from_name`.

**Comment by overlookmotel** [comment]
Closed in #5480.
